embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/SpansRepository.kt

```diff
@@ -0,0 +1,62 @@
+package io.embrace.android.embracesdk.internal.spans
+
+import io.embrace.android.embracesdk.spans.EmbraceSpan
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Allows the tracking of [EmbraceSpan] instances so that their references can be retrieved with its associated spanId
+ */
+internal class SpansRepository {
```

Comment:
I'm still missing a strategy to clear the tracked spans because you can theoretically use as parents spans that are already sent off device. I'm going to do some profiling to see how much memory keeping track of 1000+ spans take, but I'm likely going to add a facility to clear the tracked spans at some point.